engine/src/flutter/lib/web_ui/lib/src/engine/semantics/scrollable.dart

```diff
@@ -183,64 +178,47 @@ class SemanticScrollable extends SemanticRole {
     }
   }
 
-  /// Resets the scroll position (top or left) to the neutral value.
-  ///
-  /// The scroll position of the scrollable HTML node that's considered to
-  /// have zero offset relative to Flutter's notion of scroll position is
-  /// referred to as "neutral scroll position".
-  ///
-  /// We always set the scroll position to a non-zero value in order to
-  /// be able to scroll in the negative direction. When scrollTop/scrollLeft is
-  /// zero the browser will refuse to scroll back even when there is more
-  /// content available.
-  void _neutralizeDomScrollPosition() {
+  void _updateScrollableState() {
     // This value is arbitrary.
-    const int canonicalNeutralScrollPosition = 10;
     final ui.Rect? rect = semanticsObject.rect;
     if (rect == null) {
       printWarning('Warning! the rect attribute of semanticsObject is null');
       return;
     }
+    final double? scrollExtentMax = semanticsObject.scrollExtentMax;
+    final double? scrollExtentMin = semanticsObject.scrollExtentMin;
+    assert(scrollExtentMax != null);
+    assert(scrollExtentMin != null);
+    final double scrollExtentTotal =
+        scrollExtentMax! -
+        scrollExtentMin! +
+        (semanticsObject.isVerticalScrollContainer ? rect.height : rect.width);
+    // Place the _scrollOverflowElement at the beginning of the content
+    // and size it based on the total scroll extent so the browser
+    // knows how much scrollable content there is. The cross axis size
+    // should be non-zero so it is taken into account in the scrollable
+    // elements scrollHeight.
```

Comment:
nit: I'd place this comment immediately above the line that sets 1px. Otherwise, it's hard to make the connection between "non-zero cross axis" and `width = '1px'`/`height = '1px'`.